File: app/src/main/java/org/tesira/civic/db/Converters.java
package org.tesira.civic.db;

import androidx.annotation.Nullable;
import androidx.room.TypeConverter;

import org.tesira.civic.db.CardColor;

public class Converters {

    @TypeConverter
    @Nullable
    public static CardColor toCardColor(@Nullable String value) {
        if (value == null) {
            return null;
        }
        try {
            return CardColor.valueOf(value.toUpperCase());
        } catch (IllegalArgumentException e) {
            System.err.println("Ungültiger CardColor-String aus DB: '" + value + "' - wird zu null konvertiert.");
            return null;
        }
    }

    @TypeConverter
    @Nullable
    public static String fromCardColor(@Nullable CardColor color) { // Das Eingabe-Enum kann null sein
        return color == null ? null : color.name();
    }

}

File: app/src/main/kotlin/org/tesira/civic/db/Converters.kt
package org.tesira.civic.db

import androidx.room.TypeConverter
import java.util.Locale

object Converters {
    @TypeConverter
    fun toCardColor(value: String?): CardColor? {
        if (value == null) {
            return null
        }
        try {
            return CardColor.valueOf(value.uppercase(Locale.getDefault()))
        } catch (_: IllegalArgumentException) {
            System.err.println("Ungültiger CardColor-String aus DB: '$value' - wird zu null konvertiert.")
            return null
        }
    }

    @TypeConverter
    fun fromCardColor(color: CardColor?): String? {
        return color?.name
    }
}